apps/designer/app/dashboard/project-card.tsx

```diff
@@ -0,0 +1,292 @@
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  IconButton,
+  css,
+  Flex,
+  Text,
+  theme,
+  toast,
+} from "@webstudio-is/design-system";
+import { MenuIcon } from "@webstudio-is/icons";
+import type { DashboardProject } from "@webstudio-is/prisma-client";
+import { KeyboardEvent, useEffect, useRef, useState } from "react";
+import { designerPath, getPublishedUrl } from "~/shared/router-utils";
+import { Link as RemixLink, useFetcher } from "@remix-run/react";
+import { isFeatureEnabled } from "@webstudio-is/feature-flags";
+
+const projectCardContainerStyle = css({
+  overflow: "hidden",
+  width: theme.spacing[31],
+  height: theme.spacing[29],
+  borderWidth: 1,
+  borderStyle: "solid",
+  borderColor: theme.colors.borderMain,
+  borderRadius: theme.borderRadius[4],
+  background: "linear-gradient(0deg, #FBF8FF 0%, #C7C7C7 100%)",
+  "&:hover, &:focus-within": {
+    boxShadow: theme.shadows.brandElevationBig,
+  },
+});
+
+const projectCardFooterStyle = css({
+  background: theme.colors.brandBackgroundProjectCardTextArea,
+  height: theme.spacing[17],
+  py: theme.spacing[5],
+  px: theme.spacing[7],
+});
+
+const projectNameAvatarStyle = css({
+  fontFamily: theme.fonts.manrope,
+  fontWeight: 200,
+  fontSize: 360,
+  alignSelf: "center",
+  letterSpacing: "-0.05em",
+  marginLeft: "-0.5em",
+  background: "linear-gradient(0deg, #FBF8FF 25%, #E2E2E2 66%)",
+  WebkitBackgroundClip: "text",
+  backgroundClip: "text",
+  color: "transparent",
+  // @todo figure out the effect from figma in text-shadow
+  //textShadow: "0px 5px 3px rgba(251, 248, 255, 1)",
+  userSelect: "none",
+  outline: "none",
+  "&:hover, &:focus": {
+    fontWeight: 800,
+    transition: "100ms",
+  },
+});
+
+// My Next Project > MN
+const getAbbreviation = (title: string) =>
+  title
+    .split(/\s+/)
+    .slice(0, 2)
+    .map((word) => word.charAt(0).toUpperCase())
+    .join("");
+
+const usePublishedLink = ({ domain }: { domain: string }) => {
+  const [url, setUrl] = useState<URL>();
+
+  useEffect(() => {
+    // It uses `location` to detect the default values at localhost.
+    setUrl(new URL(getPublishedUrl(domain)));
+  }, [domain]);
+
+  return { url };
```

Comment:
the problem with getPublishedUrl is that it uses location when there is no env variable defined, like its the case with local env, so when it renders on that backend - browser location doesn't exist and it will mismatch at hydration

basically it boils down to having a localhost:3000 predefined by default serverside and since I haven't found a way to have it without hardcoding and without forcing a user to define it, we detect it on the client

we need a way to get the default host:port without forcing user to set it, remix can be searching for a free port if 3000 is taken, so its not guaranteed, if we hardcode the port we will run into a problem when its taken.

we basically need to get the final port from remix after it has found a free one.